test/e2e/utils/datainserter.go

```diff
@@ -32,6 +33,11 @@ type TestData struct {
 }
 
 func NewDataInserter(hosts []string) (*DataInserter, error) {
+	// Instead of specifying hosts for the provided datacenter, use 'replication_factor' as a single key to specify a default RF.
+	return NewMultiDCDataInserter(map[string][]string{"replication_factor": hosts})
```

Comment:
That's why I've put the comment there, it's somewhat hacky but avoids modifying the existing tests

Reply:
Alternatively, the `NewMultiDCDataInserter` could take a flat slice of hosts as a first argument and `map[string]int` with replication factor as a second. A bit redundant and leaking the implementation, but I don't see any other options atm. Wdyt?

Reply:
Just don't call it `replication_factor` when it means a DC and the value is "hosts". (you later use the value to compute replication factor at some place but that shouldn't be the name of the key. In the other func, the key in that map means a datacenter, so let's put in a dummy key for a DC instead, like `_datacenter` or something and let the comment say, that the key doesn't matter as it's ignored by the func.)

Reply:
The keys are used for specifying the replication_factor though. You can either specify RF per DC `"DC1": 3, "DC2": 6` (same for single DC `"DC": 3`) or the default for all DCs `replication_factor: 3`. See https://opensource.docs.scylladb.com/stable/cql/ddl.html#networktopologystrategy.

So in case a single slice of hosts is provided, I use `replication_factor` as a key to set a default. The logic can be pushed down, but it'll still have to discard the dummy datacenter name and replace it with the default.

Reply:
ok, now I see where the bad API is coming from - it's mixing two schemas into single field, making one key to be special and likely breaking if you actually name on of the DCs `replication_factor`.

given the 2 schemas are merged into one, the variables didn't make sense, so say  `for dc, hosts := range dcHosts {` would need to be `for dcOrRFMagicValue, hosts := range dcHosts {`, so I'll leave it be. thanks